recipes/physx/4.1.1/conanfile.py

```diff
@@ -0,0 +1,298 @@
+        try:
+            os.rename(extracted_dir, self._source_subfolder)
+        except OSError:
+            # workaround for permission denied on windows
+            time.sleep(10)
+            os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        tools.replace_in_file(os.path.join(self._source_subfolder, "pxshared", "include", "foundation", "PxPreprocessor.h"),
+                              "#error Exactly one of NDEBUG and _DEBUG needs to be defined!",
+                              "// #error Exactly one of NDEBUG and _DEBUG needs to be defined!")
+
+        # Comment out hard-coded PIC settings
+        tools.replace_in_file(os.path.join(self._source_subfolder, "physx", "source", "compiler", "cmake", "CMakeLists.txt"),
+                              "SET(CMAKE_POSITION_INDEPENDENT_CODE ON)",
+                              "SET(CMAKE_POSITION_INDEPENDENT_CODE ${PHYSX_CONAN_FPIC})")
+        for cmake in (
+                "FastXml.cmake",
+                os.path.join("linux", "LowLevel.cmake"),
+                os.path.join("linux", "PhysXCharacterKinematic.cmake"),
+                "LowLevel.cmake",
+                "LowLevelAABB.cmake",
+                "LowLevelDynamics.cmake",
+                "PhysX.cmake",
+                "PhysXCharacterKinematic.cmake",
+                "PhysXCommon.cmake",
+                "PhysXCooking.cmake",
+                "PhysXExtensions.cmake",
+                "PhysXFoundation.cmake",
+                "PhysXPvdSDK.cmake",
+                "PhysXTask.cmake",
+                "PhysXVehicle.cmake",
+                "SceneQuery.cmake",
+                "SimulationController.cmake",
+        ):
+            target, _ = os.path.splitext(os.path.basename(cmake))
+            tools.replace_in_file(os.path.join(self._source_subfolder, "physx", "source", "compiler", "cmake", cmake),
+                                  "SET_TARGET_PROPERTIES({} PROPERTIES POSITION_INDEPENDENT_CODE TRUE)".format(target),
+                                  "SET_TARGET_PROPERTIES({} PROPERTIES POSITION_INDEPENDENT_CODE ${{PHYSX_CONAN_FPIC}})".format(target))
+
+        for cmake_os in ("linux", "mac", "android", "ios"):
+            tools.replace_in_file(os.path.join(self._source_subfolder, "physx", "source", "compiler", "cmake", cmake_os, "CMakeLists.txt"),
+                                  "-Werror", "")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self, build_type=self._get_physx_build_type())
+
+        if self.settings.os != "Windows" and not self.options.shared:
+            self._cmake.definitions["PHYSX_CONAN_FPIC"] = self.options.fPIC
```

Comment:
```suggestion
            self._cmake.definitions["PHYSX_CONAN_FPIC"] = self.options.shared or self.options.fPIC
```

Reply:
Build failed because it didn't use proper commit I guess (same issuethan https://github.com/conan-io/conan-center-index/pull/829).

Reply:
No, it failed because `fPIC` is also required when shared.

Reply:
It's ON by default in cmake wrapper.